QuantConnect.Polygon/PolygonDataDownloader.cs

```diff
@@ -74,35 +75,64 @@ public IEnumerable<BaseData> Get(DataDownloaderGetParameters parameters)
 
             if (symbol.IsCanonical())
             {
-                using var dataQueue = new BlockingCollection<BaseData>();
-                var symbols = GetOptions(symbol, startUtc, endUtc);
+                return GetBaseData(symbol, startUtc, endUtc, dataType, resolution, exchangeHours, dataTimeZone, tickType);
+            }
+            else
+            {
+                var historyRequest = new HistoryRequest(startUtc, endUtc, dataType, symbol, resolution, exchangeHours, dataTimeZone, resolution,
+                    true, false, DataNormalizationMode.Raw, tickType);
 
-                Task.Run(() => Parallel.ForEach(symbols, targetSymbol =>
-                {
-                    var historyRequest = new HistoryRequest(startUtc, endUtc, dataType, targetSymbol, resolution, exchangeHours, dataTimeZone,
-                        resolution, true, false, DataNormalizationMode.Raw, tickType);
-                    foreach (var data in _historyProvider.GetHistory(historyRequest))
-                    {
-                        dataQueue.Add(data);
-                    }
-                })).ContinueWith(_ =>
-                {
-                    dataQueue.CompleteAdding();
-                });
+                var historyData = _historyProvider.GetHistory(historyRequest);
 
-                foreach (var data in dataQueue.GetConsumingEnumerable())
+                if (historyData == null)
                 {
-                    yield return data;
+                    return null;
                 }
+
+                return GetBaseData(historyData);
             }
-            else
+        }
+
+        private IEnumerable<BaseData>? GetBaseData(Symbol symbol, DateTime startUtc, DateTime endUtc, Type dataType, Resolution resolution, SecurityExchangeHours exchangeHours, DateTimeZone dataTimeZone, TickType tickType)
```

Comment:
I think it is a bit unclear that this method does by just looking at its name an signature. Perhaps `GetCanonicalOptionHistory` would be clearer.

Reply:
Thanks for tip.